Mathlib/Data/Vector/Defs.lean

```diff
@@ -69,6 +108,22 @@ def tail : Vector α n → Vector α (n - 1)
   | ⟨[], h⟩ => ⟨[], congrArg pred h⟩
   | ⟨_ :: v, h⟩ => ⟨v, congrArg pred h⟩
 
+@[simp] theorem toVector_tail :
+    ∀ {n} (v : Vector α n), (v.tail).toVector = v.toVector.tail
+  | 0, ⟨[], h⟩ => _root_.Vector.toList_inj.mp rfl
+  | _ + 1, ⟨_ :: v, h⟩ => _root_.Vector.toList_inj.mp <| by
+    unfold tail _root_.Vector.tail
+    simp_rw [toVector_mk, zero_lt_succ, dite_true, Vector.eraseIdx_mk,
+      eraseIdx_toArray, eraseIdx_zero, tail_cons]
+
+@[simp] theorem ofVector_tail :
+    ∀ {n} (v : _root_.Vector α n), ofVector v.tail = (ofVector v).tail
+  | 0, ⟨⟨[]⟩, _⟩ => Subtype.ext rfl
+  | _ + 1, ⟨⟨_ :: v⟩, h⟩ => Subtype.ext <| by
+    unfold tail _root_.Vector.tail
+    simp_rw [Nat.add_one_sub_one, zero_lt_succ, dite_true, Vector.eraseIdx_mk,
+      eraseIdx_toArray, eraseIdx_zero, tail_cons, ofVector_mk]
```

Comment:
I would assume you can more directly prove these with a result about `tail` on lists vs arrays?

Reply:
You would think this but `Array.tail` is not defined and `Vector.tail` has quite an ugly definition. I couldn't find a neater result here as I could for `append`, say.